src/pkg/scheduler/callback.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scheduler
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/goharbor/harbor/src/jobservice/job"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/pkg/task"
+)
+
+var (
+	registry = make(map[string]CallbackFunc)
+)
+
+// CallbackFunc defines the function that the scheduler calls when triggered
+type CallbackFunc func(interface{}) error
+
+func init() {
+	if err := task.RegisterCheckInProcessor(JobNameScheduler, triggerCallback); err != nil {
+		log.Errorf("failed to register check in processor for scheduler: %v", err)
+	}
+}
+
+// RegisterCallbackFunc registers the callback function which will be called when the scheduler is triggered
+func RegisterCallbackFunc(name string, callbackFunc CallbackFunc) error {
```

Comment:
If the callback func can be registered with the job name, that means each kind of job has only one callback func, why we need to persist the callback name and params into the DB?

Reply:
Resolved as we had an offline discussion